packages/connector-chromium/src/connector.ts

```diff
@@ -0,0 +1,534 @@
+     * * uses `favicon.ico` and the final url after redirects.
+     */
+    private async getFavicon(dom: HTMLDocument) {
+        const element = (await dom.querySelectorAll('link[rel~="icon"]'))[0];
+        const href = (element && element.getAttribute('href')) || '/favicon.ico';
+
+        try {
+            debug(`resource ${href} to be fetched`);
+            const fullFaviconUrl = this._finalHref + href.substr(1);
+
+            await this._engine.emitAsync('fetch::start', { resource: fullFaviconUrl });
+
+            const content = await this.fetchContent(new URL(fullFaviconUrl));
+
+            const data: FetchEnd = {
+                element: null,
+                request: content.request,
+                resource: content.response.url,
+                response: content.response
+            };
+
+            await this._engine.emitAsync('fetch::end::image', data);
+        } catch (error) {
+            const event: FetchError = {
+                element,
+                error,
+                hops: [],
+                resource: href
+            };
+
+            await this._engine.emitAsync('fetch::error', event);
+        }
+    }
+
+    private getRawResponse(response: puppeteer.Response) {
+        const connector = this; // eslint-disable-line
+
+        return async function (this: any) {
+
+            const that = this; // eslint-disable-line
+
+            if (that._rawResponse) {
+                return that._rawResponse;
+            }
+
+            const rawContent = await response.buffer();
+            const responseHeaders = response.headers();
+
+            if (rawContent && rawContent.length.toString() === responseHeaders['Content-Length']) {
+                // Response wasn't compressed so both buffers are the same
+                return rawContent;
+            }
+
+            const requestHeaders = response.request().headers();
+            const responseUrl = response.url();
+
+            /*
+             * Real browser connectors automatically request using HTTP2. This spec has
+             * [`Pseudo-Header Fields`](https://tools.ietf.org/html/rfc7540#section-8.1.2.3):
+             * `:authority`, `:method`, `:path` and `:scheme`.
+             *
+             * An example of request with those `Pseudo-Header Fields` to google.com:
+             *
+             * ```
+             * :authority:www.google.com
+             * :method:GET
+             * :path:/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png
+             * :scheme:https
+             * accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*\/*;q=0.8
+             * accept-encoding:gzip, deflate, br
+             * ...
+             * ```
+             *
+             * The `request` module doesn't support HTTP2 yet: https://github.com/request/request/issues/2033
+             * so the request need to be transformed to valid HTTP 1.1 ones, basically removing those headers.
```

Comment:
This can go away once we move off of `request`, right?

Reply:
[`got` has some experimental HTTP2 support](https://www.npmjs.com/package/got#experimental-http2-support). Once it works I think we can remove it. I'm no expert on HTTP2 so maybe some headers will still need to be removed.